internal/core/src/query/visitors/ExecPlanNodeVisitor.cpp

```diff
@@ -135,23 +135,35 @@ ExecPlanNodeVisitor::visit(RetrievePlanNode& node) {
 
     auto active_count = segment->get_active_count(timestamp_);
 
-    if (active_count == 0) {
+    if (active_count == 0 && !node.is_count) {
         retrieve_result_opt_ = std::move(retrieve_result);
         return;
     }
 
-    BitsetType bitset_holder;
-    if (node.predicate_ != nullptr) {
+    BitsetType bitset_holder(active_count);
+    if (node.predicate_.has_value() && node.predicate_.value() != nullptr) {
         bitset_holder = ExecExprVisitor(*segment, active_count, timestamp_)
-                            .call_child(*(node.predicate_));
+                            .call_child(*(node.predicate_.value()));
         bitset_holder.flip();
     }
 
     segment->mask_with_timestamps(bitset_holder, timestamp_);
 
     segment->mask_with_delete(bitset_holder, active_count, timestamp_);
     // if bitset_holder is all 1's, we got empty result
-    if (bitset_holder.all()) {
+    if (bitset_holder.all() && !node.is_count) {
```

Comment:
When biset_holder.all() is true, add logic for fast return when is_count is True

Reply:
In fact, we immediately return the count result at line 159.